src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -1418,6 +1418,122 @@ def aks_upgrade(cmd, client, resource_group_name, name, kubernetes_version, no_w
     return sdk_no_wait(no_wait, client.create_or_update, resource_group_name, name, instance)
 
 
+def aks_use_devconnect(cmd, client, cluster_name, resource_group_name, space_name='default'):  # pylint: disable=line-too-long
+    """
+    Use Azure Dev Connect with a managed Kubernetes cluster.
+
+    :param cluster_name: Name of the target AKS cluster.
+    :type cluster_name: String
+    :param resource_group_name: Name of the target AKS cluster's resource group.
+    :type resource_group_name: String
+    :param space_name: The isolated space in the cluster to develop in.
+    :type space_name: String
+    """
+
+    vsce_tool = 'Visual Studio Connected Development Services'
+    vsce_cli = 'vsce'
+    install_vsce = False
+    system = platform.system()
+    if system == 'Linux':
+        # Linux
+        raise CLIError('Linux is not supported yet.')
+    elif system == 'Windows':
+        # Windows
+        # Dev Connect Install Path (WinX)
+        vsce_cli = "C:\\Program Files\\Microsoft SDKs\\Azure\\Visual Studio Connected Environment CLI\\vsce.exe"
+        setup_file = 'vsce-winx-setup.exe'
+        setup_url = "https://aka.ms/get-vsce-windows-az"
+        setup_args = [setup_file]
+    elif system == 'Darwin':
+        # OSX
+        setup_file = 'vsce-osx-setup.sh'
+        setup_url = "https://aka.ms/get-vsce-osx-az"
+        setup_args = ['bash', setup_file]
+
+    try:
+        _ensure_dev_connected_installed(vsce_cli)
+    except OSError:
+        install_vsce = True
+
+    if install_vsce:
+        # Install VSCE
+        logger.info('Installing Dev Connect commands...')
+        import urllib.request
+        urllib.request.urlretrieve(setup_url, setup_file)
+        try:
+            subprocess.call(
+                setup_args, universal_newlines=True, stdin=None, stdout=None, stderr=None, shell=False)
```

Comment:
I do not believe `subprocess.call` can raise a `subprocess.CalledProcessError` according to this - https://docs.python.org/3/library/subprocess.html#subprocess.call.

This should probably be `subprocess.check_call` instead.

Reply:
I see. In the same code base if you check, subprocess.CalledProcessError is used for 2 types of calls: subprocess.check_output and subprocess.call. I referred to those while making my changes.

What you say makes sense, but I guess then the calls for 'helm install' and 'helm del' should also be updated to check the retCode and not try to catch CalledProcessError